File: app/src/main/java/com/absinthe/anywhere_/ui/settings/IntervalDialogFragment.java
package com.absinthe.anywhere_.ui.settings;

import android.app.Dialog;
import android.os.Bundle;

import androidx.annotation.NonNull;

import com.absinthe.anywhere_.R;
import com.absinthe.anywhere_.model.GlobalValues;
import com.absinthe.anywhere_.view.AnywhereDialogBuilder;
import com.absinthe.anywhere_.view.AnywhereDialogFragment;
import com.absinthe.anywhere_.viewbuilder.entity.IntervalDialogBuilder;

public class IntervalDialogFragment extends AnywhereDialogFragment {
    private IntervalDialogBuilder mBuilder;

    @NonNull
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        AnywhereDialogBuilder builder = new AnywhereDialogBuilder(getContext());
        mBuilder = new IntervalDialogBuilder(getContext());
        mBuilder.slider.setValue(GlobalValues.sDumpInterval / 1000f);

        return builder.setView(mBuilder.getRoot())
                .setTitle(R.string.dialog_set_interval_title)
                .setPositiveButton(R.string.dialog_delete_positive_button, (dialog, which) -> {
                    int interval = (int) mBuilder.slider.getValue() * 1000;
                    GlobalValues.setsDumpInterval(interval);
                })
                .setNegativeButton(R.string.dialog_delete_negative_button, null)
                .create();
    }
}

File: app/src/main/java/com/absinthe/anywhere_/ui/settings/IntervalDialogFragment.kt
package com.absinthe.anywhere_.ui.settings

import android.app.Dialog
import android.content.DialogInterface
import android.os.Bundle
import com.absinthe.anywhere_.R
import com.absinthe.anywhere_.constants.GlobalValues
import com.absinthe.anywhere_.view.app.AnywhereDialogBuilder
import com.absinthe.anywhere_.view.app.AnywhereDialogFragment
import com.absinthe.anywhere_.viewbuilder.entity.IntervalDialogBuilder

class IntervalDialogFragment : AnywhereDialogFragment() {

    private lateinit var mBuilder: IntervalDialogBuilder

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        mBuilder = IntervalDialogBuilder(requireContext()).apply {
            slider.value = when {
                GlobalValues.dumpInterval / 1000f > slider.valueTo -> {
                    slider.valueTo
                }
                GlobalValues.dumpInterval / 1000f < slider.valueFrom -> {
                    slider.valueFrom
                }
                else -> {
                    GlobalValues.dumpInterval / 1000f
                }
            }
        }

        return AnywhereDialogBuilder(requireContext()).setView(mBuilder.root)
                .setTitle(R.string.dialog_set_interval_title)
                .setPositiveButton(R.string.dialog_delete_positive_button) { _: DialogInterface?, _: Int ->
                    GlobalValues.dumpInterval = mBuilder.slider.value.toInt() * 1000
                }
                .setNegativeButton(R.string.dialog_delete_negative_button, null)
                .create()
    }
}